sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-text.models.ts

```diff
@@ -0,0 +1,124 @@
+import { Contains, IsDate, IsNumberString, Length } from "class-validator";
+import { APPLICATION_NUMBER_LENGTH } from "@sims/sims-db";
+import { IsValidSIN } from "../../utilities/class-validation";
+import { getDateOnlyFromFormat } from "@sims/utilities";
+
+export const DATE_FORMAT = "YYYYMMDD";
+/**
+ * user-friendly header names used in the text to be populated by the user.
+ * The text model parser uses this as a base to parse the text string into an object model.
+ */
+export const DataTextHeaders = {
+  recordType: "Record Type",
+  sin: "SIN",
+  applicationNumber: "Application Number",
+  withdrawalDate: "Withdrawal Date",
+};
+
+export enum RecordType {
+  ApplicationBulkWithdrawalHeaderRecordType = "1",
+  ApplicationBulkWithdrawalDataRecordType = "2",
+  ApplicationBulkWithdrawalFooterRecordType = "3",
+}
+
+export class ApplicationWithdrawalTextModel {
+  /**
+   * Data record type.
+   */
+  @Contains(RecordType.ApplicationBulkWithdrawalDataRecordType, {
+    message: `${DataTextHeaders.recordType} must be valid.`,
+  })
+  recordType: RecordType;
+  /**
+   * SIN.
+   */
+  @IsValidSIN({
+    message: `${DataTextHeaders.sin} must be a valid SIN.`,
+  })
+  sin: string;
+  /**
+   * Application Number.
+   */
+  @IsNumberString(
+    { no_symbols: true },
+    {
+      message: `${DataTextHeaders.applicationNumber} must be a valid numeric application number.`,
+    },
+  )
+  @Length(APPLICATION_NUMBER_LENGTH, APPLICATION_NUMBER_LENGTH, {
+    message: `${DataTextHeaders.applicationNumber} must be a valid application number.`,
+  })
+  applicationNumber: string;
+  /**
+   * Withdrawal Date.
+   */
+  @IsDate({
+    message: `${DataTextHeaders.withdrawalDate} must be a valid withdrawal date.`,
+  })
+  withdrawalDate: Date;
+}
+
+/**
+ * Represents the validation performed on a text model.
+ */
+export interface ApplicationWithdrawalTextValidationResult {
+  /**
+   * Zero base index of the record in the list of the records.
+   * Does not consider possible header.
+   */
+  index: number;
+  /**
+   * Model that was validated.
+   */
+  textModel: ApplicationWithdrawalTextModel;
+  /**
+   * List of possible errors. If no error is present it
+   * means the model was successfully validated.
+   */
+  errors: string[];
+}
+
+export class ApplicationBulkWithdrawalHeader {
```

Comment:
As per our call with devs, for the story we decided not to create models like the way we do for file integration.

The outcome of the discussion was follow exactly same way as of offering bulk upload. the only difference here is using a custom transformation to transform the flatfile string to model[] where offering bulk was using papa parse framework to parse a csv file.

Reply:
I recall the discussion, its more like trying to use a string to model transformation, if possible. I thought of writing the actual model transformations, as we have custom header and footer validations, and using a parser would also be an overhead for the 3 fields we try to validate. So went with this approach.

Reply:
Resolving as dicussed.